File: apps/foo/src/main/java/smecalculus/bezmen/messaging/impl/springmvc/SepulkaController.java
package smecalculus.bezmen.messaging.impl.springmvc;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import smecalculus.bezmen.messaging.spec.SepulkaClient;
import smecalculus.bezmen.messaging.spec.SepulkaMsgMapper;
import smecalculus.bezmen.messaging.spec.SepulkaRegReq;
import smecalculus.bezmen.messaging.spec.SepulkaRegReqMsg;
import smecalculus.bezmen.messaging.spec.SepulkaRegRes;
import smecalculus.bezmen.messaging.spec.SepulkaRegResMsg;

@RestController
@RequestMapping("sepulkas")
public record SepulkaController(SepulkaClient client, SepulkaMsgMapper mapper) {
    @PostMapping
    ResponseEntity<SepulkaRegResMsg> register(@RequestBody SepulkaRegReqMsg sepulkaRegReqMsg) {
        SepulkaRegReq sepulkaRegReq = mapper.toDomain(sepulkaRegReqMsg);
        SepulkaRegRes sepulkaRegRes = client.register(sepulkaRegReq);
        return ResponseEntity.status(HttpStatus.CREATED).body(mapper.toMsg(sepulkaRegRes));
    }
}

File: apps/sepuling-kotlin/src/main/kotlin/smecalculus/bezmen/messaging/springmvc/SepulkaController.kt
package smecalculus.bezmen.messaging.springmvc

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import smecalculus.bezmen.messaging.MessageEm
import smecalculus.bezmen.messaging.SepulkaClient

@RestController
@RequestMapping("sepulkas")
class SepulkaController(
    private val client: SepulkaClient
) {
    @PostMapping
    fun register(@RequestBody requestEdge: MessageEm.RegistrationRequest): ResponseEntity<MessageEm.RegistrationResponse> {
        val responseEdge = client.register(requestEdge)
        return ResponseEntity.status(HttpStatus.CREATED).body(responseEdge)
    }
}
